### PR TITLE
feat(GRPC): Update API for Suggestion

### LINKED ISSUE #507: Redesign Suggestion service design
Proposal is to make new Suggestion CR  to serve suggestion requests from the experiment controller. Currently, experiment controller directly talks to suggestion service and gets suggestions. 

In the new design,
Experiment controller creates Suggestion CR for each experiment to get suggestion results.Suggestion controller picks it up and talks to right algorithm service(as currently) and updates the Suggestion CR status with suggestion results and state. Experiment controller reads the suggestion results from CR and creates trials.

With new Suggestion CR and controller implementation,
1. It is more K8s native as DB dependency is avoided
2. It can take care of long running experiments. Suggestion CR is created per experiment by the experiment controller.
3. State or algorithm settings(key-value pair) will be stored in the CR instead of external DB.  Suggestion controller can thus send just the required information to each algorithm service.

/assign
/cc @gaocegege @richardsliu @hougangliu  @YujiOshima 

/kind feature

Related: #463 #471 #500

### PR DESCRIPTION
**Pull request by gaocegege**
**What this PR does / why we need it**:

This PR is to update the GRPC API, to avoid DB in the main workflow and remove manager requests from the suggestion side.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref #507 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/743)

### CODE REVIEW

File: pkg/controller.v1alpha3/experiment/suggestion/suggestion.go [hunk 1]
```diff
@@ -18,14 +17,15 @@ func New() Suggestion {
...
+	// 	ExperimentName: instance.Name,
+	// 	AlgorithmName:  instance.Spec.Algorithm.AlgorithmName,
+	// 	RequestNumber:  addCount,
+	// }
+	// if reply, err := common.GetSuggestions(request); err != nil {
+	// 	return nil, err
+	// } else {
+	// 	return reply.Trials, nil
+	// }
+	return nil, nil
```
Comment (gaocegege) [inline pkg/controller.v1alpha3/experiment/suggestion/suggestion.go]: This is to make the compilation succeeded.

File: pkg/apis/manager/v1alpha3/api.proto [hunk 2]
```diff
@@ -516,13 +516,14 @@ message GetObservationLogReply {
...
-    string algorithm_name = 2;
+    Experiment experiment = 1;
+    repeated Trial trials = 2; // all completed trials owned by the experiment.
     int32 request_number = 3; ///The number of Suggestion you request at one time. When you set 3 to request_number, you can get three Suggestions at one time.
 }
 
 message GetSuggestionsReply {
-    repeated Trial trials = 1;
+    repeated Trial trials = 1; // trials should be created in the next run.
+    AlgorithmSpec algorithm = 2;
```
Comment (johnugeorge) [inline pkg/apis/manager/v1alpha3/api.proto]: How do we intend to use AlgorithmSpec in the reply?

Comment (gaocegege) [reply pkg/apis/manager/v1alpha3/api.proto]: It is used to support hyperband. If the algorithm wants to maintain some states using algorithm settings, I think we should support it in the current version. In the future, we should remove it.

**Comment by gaocegege** [comment]
@johnugeorge I think we cannot only update the GRPC API. CI will fail.

Can you update API, manager implementation and controllers in one PR?

File: pkg/apis/manager/v1alpha3/api.proto [hunk 3]
```diff
@@ -516,13 +516,14 @@ message GetObservationLogReply {
...
-    string experiment_name = 1;
-    string algorithm_name = 2;
+    Experiment experiment = 1;
+    repeated Trial trials = 2; // all completed trials owned by the experiment.
     int32 request_number = 3; ///The number of Suggestion you request at one time. When you set 3 to request_number, you can get three Suggestions at one time.
 }
 
 message GetSuggestionsReply {
-    repeated Trial trials = 1;
+    repeated Trial trials = 1; // trials should be created in the next run.
```
[the remark this answers is not in the document]
Comment (gaocegege) [reply pkg/apis/manager/v1alpha3/api.proto]: I think we can fix it in another PR.

**Pull request merged**